paymail_service_provider.go

```diff
@@ -173,6 +175,15 @@ func (p *PaymailDefaultServiceProvider) RecordTransaction(ctx context.Context,
 		return nil, err
 	}
 
+	if p2pTx.DecodedBeef == nil {
+		return &paymail.P2PTransactionPayload{
+			Note: p2pTx.MetaData.Note,
+			TxID: transaction.ID,
+		}, nil
+	}
+
+	saveBEEFTxInputs(ctx, p.client, p2pTx, rts)
```

Comment:
We shouldn't wait with response for saving these transactions,  please run it in separate go routine